api/src/pcapi/core/search/backends/algolia.py

```diff
@@ -148,6 +220,38 @@ def pop_venue_ids_from_queue(self, count: int, from_error_queue: bool = False) -
             logger.exception("Could not pop offer ids to index from queue", extra={"queue": redis_set_name})
             return set()
 
+    def pop_collective_offer_ids_from_queue(self, count: int, from_error_queue: bool = False) -> set[int]:
+        if from_error_queue:
+            redis_set_name = REDIS_COLLECTIVE_OFFER_IDS_IN_ERROR_TO_INDEX
+        else:
+            redis_set_name = REDIS_COLLECTIVE_OFFER_IDS_TO_INDEX
+        try:
+            collective_offer_ids = self.redis_client.spop(redis_set_name, count)
+            return {int(collective_offer_id) for collective_offer_id in collective_offer_ids}  # str -> int
+        except redis.exceptions.RedisError:
+            logger.exception(
+                "Could not pop collective offer ids to index from queue",
+                extra={"queue": redis_set_name},
+            )
+            return set()
+
+    def pop_collective_offer_template_ids_from_queue(self, count: int, from_error_queue: bool = False) -> set[int]:
+        if from_error_queue:
+            redis_set_name = REDIS_COLLECTIVE_OFFER_TEMPLATE_IDS_IN_ERROR_TO_INDEX
+        else:
+            redis_set_name = REDIS_COLLECTIVE_OFFER_TEMPLATE_IDS_TO_INDEX
+        try:
+            collective_offer_template_ids = self.redis_client.spop(redis_set_name, count)
+            return {
+                int(collective_offer_template_id) for collective_offer_template_id in collective_offer_template_ids
+            }  # str -> int
```

Comment:
(style) en renommage la variable, ça tient sur 1 ligne, donc plus lisible !

```suggestion
            return {int(template_id) for template_id in collective_offer_template_ids}  # str -> int
```

Cette suggestion vous est proposée par Jean-Michel Chipoteur. De rien ! :)